contracts/test/ocr2/sequencer-uptime-feed.ts

```diff
@@ -0,0 +1,121 @@
+import { expect } from 'chai'
+import { starknet } from 'hardhat'
+import { StarknetContract, StarknetContractFactory, Account } from 'hardhat/types/runtime'
+import { number } from 'starknet'
+import { assertErrorMsg } from '../utils'
+
+describe('SequencerUptimeFeed test', function () {
+  this.timeout(300_000)
+
+  let account: Account
+  let nonOwner: Account
+
+  // should be beforeeach, but that'd be horribly slow. Just remember that the tests are not idempotent
+  before(async function () {
+    account = await starknet.deployAccount('OpenZeppelin')
+    nonOwner = await starknet.deployAccount('OpenZeppelin')
+  })
+
+  describe('Test inheritence', function () {
+    const user = 101
+    let uptimeFeedContract: StarknetContract
+
+    before(async function () {
+      const uptimeFeedFactory = await starknet.getContractFactory('sequencer_uptime_feed')
+      uptimeFeedContract = await uptimeFeedFactory.deploy({
+        initial_status: 0,
+        owner_address: number.toBN(account.starknetContract.address),
+      })
+    })
+
+    it('Test grainting access', async function () {
+      await account.invoke(uptimeFeedContract, 'add_access', {
+        user,
+      })
+
+      try {
+        await nonOwner.invoke(uptimeFeedContract, 'add_access', {
+          user,
+        })
+
+        expect.fail()
+      } catch (err: any) {
+        assertErrorMsg(err?.message, 'Ownable: caller is not the owner')
+      }
+    })
+
+    it('Test has_access', async function () {
+      {
```

Comment:
This should be two separate tests.

```
describe('#has_access', () => {
  describe('when called by a user that has access', () => {
    it('returns true', async () => {
       // Add test here
    })
  })

 describe('when called by a user that has no access', () => {
    it('returns false', async () => {
       // Add test here
    })
  })
})
```

Reply:
So describe contains only one test every time? I thinks test file and log will turn into large matreshka. I would suggest having describes just for functions and then write scenarios in `it`

Reply:
Sure that’s fine as long as we are descriptive about what scenario is being tested.

Reply:
Seems like we state same thing twice:
```
describe('when called by a user that has access', () => {
    it('returns true', async () => {
```
In context of `has_access` function first and second line say the same thing.
Same for case with `false`

While that would be explicit enough in my opinion:
```
describe('#has_access', () => {
  it('when called by a user that has access', async () => {
```

Reply:
@cds95 what will we conclude here? your decision! :)

Reply:
I still prefer us to explicitly callout how the function should behave.  Saying `when called by a user that has access` doesn't really tell the reader what the expected behavior is.